Fix mpd shuffle/random status

apetrycki opened the pull request:
MPD always shows true for shuffle.  For some reason casting the 0 or 1 as a boolean does not work.  Now returns 'true' or 'false' based on the value of 'random'.

Not sure if there's a better way to do this, but it works.

apetrycki committed:
Fix shuffle/random status

MPD always shows true for shuffle.  For some reason casting the 0 or 1 as a boolean does not work.  Now returns 'true' or 'false' based on the value of 'random'.

homeassistant commented:
Hi @apetrycki, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#19308).

Once you do that we will be able to review and accept this pull request.

Thanks!

MartinHjelmare commented on homeassistant/components/media_player/mpd.py:
```diff
@@ -319,7 +319,10 @@ def play_media(self, media_type, media_id, **kwargs):
     def shuffle(self):
         """Boolean if shuffle is enabled."""
-        return bool(self._status['random'])
+        if self._status['random'] == '1':
```
```suggestion
        return bool(int(self._status['random']))
```

MartinHjelmare commented on homeassistant/components/media_player/mpd.py:
```diff
@@ -319,7 +319,10 @@ def play_media(self, media_type, media_id, **kwargs):
+        if self._status['random'] == '1':
+            return 'true'
+        else:
+            return 'false'
```
This is not a boolean. That wouldn't have worked.

apetrycki replied:
It does seem to work as it's what I've been running, but yours is a much better solution.  I'll change it and test it out.

MartinHjelmare reviewed (changes requested):
Converting to int first before bool should do the trick.

MartinHjelmare committed:
Update homeassistant/components/media_player/mpd.py

Change to correct way of returning shuffle boolean. 'random' needs to be cast as an integer before being cast as a boolean.

Co-Authored-By: apetrycki <34962392+apetrycki@users.noreply.github.com>

apetrycki committed:
Remove incorrect string code

Original fix method returns a string instead of a boolean.  Removed in favor of MartinHjelmare's method.
